htlcswitch/interceptable_switch.go

```diff
@@ -115,15 +115,16 @@ type interceptedForward struct {
 }
 
 // Packet returns the intercepted htlc packet.
-func (f *interceptedForward) Packet() lnwire.UpdateAddHTLC {
-	return *f.htlc
-}
-
-// CircuitKey returns the circuit key for the intercepted htlc.
-func (f *interceptedForward) CircuitKey() channeldb.CircuitKey {
-	return channeldb.CircuitKey{
-		ChanID: f.packet.incomingChanID,
-		HtlcID: f.packet.incomingHTLCID,
+func (f *interceptedForward) Packet() InterceptedPacket {
+	return InterceptedPacket{
+		IncomingCircuit: channeldb.CircuitKey{
+			ChanID: f.packet.incomingChanID,
+			HtlcID: f.packet.incomingHTLCID,
+		},
+		OutgoingChanID: f.packet.outgoingChanID,
```

Comment:
discussed offline: since the outgoing channel id isn't guaranteed to be the channel this HTLC is forwarded over this could lead to confusion.  imo it would be better to include the pubkey of the next hop

Reply:
It seems there is no guarantee that we can always populate the next hop pubkey. We may not know about the channel when we intercept, but when we forward later in the switch, it may be known. So I think we also need to expose the outgoing chan id?

Of course we can do a best-effort lookup, but I don't mind off-loading that to the client for now.

Reply:
i think we should at least consider changing the name from `OutgoingChanID` to something that indicates more uncertainty, e.g. `RequestedOutgoingChanID` or something like that. We discussed a bit offline, but seems we should always be able to know the public key for any active channels? and if not we fail back the HTLC?

Reply:
Made it `OutgoingRequestedChanId` so it has the same prefix as the other outgoing attributes. Decided offline to discuss exposing pubkey for a follow up.